### DISCUSSION TITLE
Lsky Pro Enterprise BUG集中反馈及功能建议

### DISCUSSION THREAD

**Discussion by Connor3773** [lsky-org/lsky-pro#556] [问答]
更新时间：2022-10-29 23:00
程序版本：Lsky Pro Enterprise V1.3
服务器：Centos 7.9 + nginx 1.22.1 + PHP 8.0.22 + Mysql 8.0.24 (utf8mb4-InnoDB) + ImageMagick 7.1.0-16 + imagick 3.7.0 (With WEBP support) 
浏览器：MS Edge 106.0.1370.52

### **Bug:**

1. 
上传一张特定图片后删除（此时用户图库为空），该用户后台500错误。需重新上传一张图片后恢复（需为其他图片，否则删除后错误复现）。

日志：
prod.ERROR: Undefined array key -1 {"view":{"view":"/www/wwwroot/img.mwuss.com/resources/views/layouts/sidebar.blade.php","data":[]},"userId":1,"exception":"[object] (Spatie\\LaravelIgnition\\Exceptions\\ViewException(code: 0): Undefined array key -1 at /www/wwwroot/img.mwuss.com/app/functions.php:131)

图片地址：https://img.mwuss.com/i/2022/10/29/635cb813b4431.jpg

2.
上传WEBP图片/设置角色组图片保存格式为WEBP后，生成缩略图时出现异常。

日志：
[previous exception] [object] (ImagickException(code: 420): no decode delegate for this image format `VIDEO' @ error/blob.c/BlobToImage/458 at /www/wwwroot/img.mwuss.com/vendor/intervention/image/src/Intervention/Image/Imagick/Decoder.php:87)
prod.ERROR: 生成缩略图时出现异常 {"file":"/www/wwwroot/img.mwuss.com/vendor/intervention/image/src/Intervention/Image/Imagick/Decoder.php","line":90,"message":"Unable to read image from binary data.","trace":"#0 /www/wwwroot/img.mwuss.com/vendor/intervention/image/src/Intervention/Image/AbstractDecoder.php(332): Intervention\\Image\\Imagick\\Decoder->initFromBinary()

图片地址：https://img.mwuss.com/i/2022/10/29/635cb8d89bc9f.webp

3.
上传图片后，上传页的缩略图链接错误，未及时更新（显示为同URL链接）。

4.
角色组设置图片保存格式后，系统统计的容量使用错误。容量被统计为原图片的大小，而非转换格式后的实际图片大小。缩略图占用的空间也未计算进去。

5.
系统控制台用户分布统计错误，只按照原始用户组进行了统计，自定义用户组未统计。

6.
系统设置-常规设置中各项，如首字符为空格，会被系统意外删除。底部代码设置项html无法正常解析。

### **功能建议：**
1. 希望用户增加余额功能。可充值余额消费。管理员可编辑余额。
2. 希望后台订单管理增加手动完成功能（手动标记为已支付并开通）
3. 希望支持多管理员功能。可修改用户为管理员还是普通用户。
4. **希望在图库中点开图片能有复制各种链接的选项，现在点开只能查看图片，想复制链接只能通过鼠标右键该图片。这导致移动端无法复制图片链接。**
5. 建议图片上传窗口设计成当前窗口中弹出式的，上传完成后可以快速关闭并留在当前页面。图库实时更新。
6. 建议增加上传完成后点击链接即可直接复制的功能，目前点击会自动全选但仍需要手动复制。
7. 建议当关闭游客上传功能时，访问网站首页不直接跳转登录页。只有当涉及需要登陆的操作再跳转。
8. 建议增加新反馈的邮件通知。

感谢作者的辛勤付出！

**Comment by 0xxb** [comment]
感谢反馈，会在下个版本中改进。

**Reply by Connor3773** [reply]
又补充了一点:)
